### PR TITLE
HARVESTER: Changes from Harvester

### PR DESCRIPTION
**Pull request by DaiYuzeng**
### Summary
There is some changes from harvester repo.

### CODE REVIEW

File: shell/components/ExplorerMembers.vue [hunk 1]
```diff
@@ -324,7 +328,7 @@ export default {
         />
       </Tab>
       <Tab
-        v-if="canManageProjectMembers"
+        v-if="canManageProjectMembers && !isHarvester"
```
Comment (DaiYuzeng) [inline shell/components/ExplorerMembers.vue:331]: Harvester want to display `Cluster Membership` only

Harvester:
image

Rancher:
image

**Comment by DaiYuzeng** [comment]
@WuJun2016 PTAL! Thanks!

**Review by richard-cox — CHANGES_REQUESTED** [review]
We need to find a better way to support disabling / hiding functionality without `if harvesters` in the dashboard code. 

@mantis-toboggan-md can you check with @aalves08 / @jordojordo if a way exists already for these places (PR probably good for 2.7q3), or how to get something introduced to support them (PR would probably be 2.8q4)?

**Comment by richard-cox** [comment]
I'm not sure when this change is required, so putting in to 2.8q4. @WuJun2016 @mantis-toboggan-md please update if this is incorrect.

File: shell/components/form/SecretSelector.vue [hunk 2]
```diff
@@ -90,7 +90,8 @@ export default {
       }
     },
     secrets() {
-      const allSecrets = this.$store.getters[`${ this.inStore }/all`](SECRET);
+      const inStore = this.$store.getters['currentProduct'].inStore;
+      const allSecrets = this.$store.getters[`${ inStore }/all`](SECRET);
```
Comment (mantis-toboggan-md) [inline shell/components/form/SecretSelector.vue:94]: The `inStore` component property no longer works with this change

**Review by mantis-toboggan-md — CHANGES_REQUESTED** [review]
There are a handful of things in this PR which I think would require additions to the extensions API to implement in a less Harvester-specific way. 
This patterns appears repeatedly in this PR:

```
const currentCluster = this.$store.getters['currentCluster'];
const inStore = currentCluster.isHarvester ? 'harvester' : 'cluster';
```

If `this.$store.getters['currentStore']` doesn't work here I think we need to revisit this in Q4 to come up with a more extensions-generic solution. 

We may be able to accomodate the Harvester-specific logging functionality by [overwriting the logging forms](https://rancher.github.io/dashboard/extensions/api/components/resources#editconfig-view) - as is we can add tabs to the logging forms but hiding existing inputs is trickier.

I think we would need to expand the extensions API to facilitate hiding the project membership tab.

**Comment by aalves08** [comment]
@mantis-toboggan-md could you create an issue labeled `area/plugins` with the areas you wished to be addressed by the extensions API?

File: shell/edit/logging-flow/index.vue [hunk 3]
```diff
@@ -51,15 +57,17 @@ export default {
   mixins: [CreateEditView],
 
   async fetch() {
-    const hasAccessToClusterOutputs = this.$store.getters[`cluster/schemaFor`](LOGGING.CLUSTER_OUTPUT);
-    const hasAccessToOutputs = this.$store.getters[`cluster/schemaFor`](LOGGING.OUTPUT);
+    const currentCluster = this.$store.getters['currentCluster'];
+    const inStore = currentCluster.isHarvester ? VIRTUAL : 'cluster';
```
Comment (mantis-toboggan-md) [inline shell/edit/logging-flow/index.vue:61]: Could this be switched to use `this.$store.getters['currentProduct'].inStore`?

Comment (WuJun2016) [reply shell/edit/logging-flow/index.vue:61]: I think here we can switch to `this.$store.getters['currentProduct'].inStore`

**Comment by torchiaf** [comment]
Closing as superseded.

**Pull request closed without merging**